src/lib/array.jl

```diff
@@ -63,6 +63,12 @@ end
   circshift(A, shifts), Δ -> (circshift(Δ, map(-, shifts)), nothing)
 end
 
+@adjoint function reverse(x::AbstractArray, args...; kwargs...)
+  _reverse(t) = reverse(t, args...; kwargs...)
+  _nothings(t) = map(_->nothing, keys(t))
+  _reverse(x), Δ->(_reverse(Δ), _nothings(args)..., _nothings(kwargs)...)
```

Comment:
You don't need to return adjoints for the kwargs here.

Reply:
Thanks, duly noted